extensions/common/gcp/gcp-core/src/main/java/org/eclipse/edc/gcp/bigquery/BigQueryDataAddress.java

```diff
@@ -0,0 +1,112 @@
+/*
+ *  Copyright (c) 2024 Google LLC
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Google LLC
+ *
+ */
+
+package org.eclipse.edc.gcp.bigquery;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+import org.eclipse.edc.spi.types.domain.DataAddress;
+
+import java.util.Optional;
+
+import static java.util.Collections.emptyMap;
+
+/**
+ * This is a wrapper class for the {@link DataAddress} object, which has typed accessors for
+ * properties specific to a BigQuery endpoint.
+ */
+@JsonTypeName()
+@JsonDeserialize(builder = DataAddress.Builder.class)
```

Comment:
Is this class supposed to be serializable? where (in what situation) is it serialized/deserialized? If so, there should be a ser/des test for it.

Reply:
Not serializable